DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/niosv/software/simple_dma_test/src/simple_dma_test.c

```diff
@@ -0,0 +1,203 @@
+//  Copyright (c) 2022 Intel Corporation
+//  SPDX-License-Identifier: MIT
+//
+// This design leverages a simple DMA oneAPI kernel that has had its RTL
+// generated and integrated into the Nios V test system that will be used to
+// control the accelerator. The DMA kernel has been configured to read from
+// memory in memory location 0 (Buffer Location 0) and write to memory in memory
+// location 1 (Buffer Location 1).
+//
+// The CPU is configured to have a single peripheral space located at
+// 0x0010_0000 and it is 1MB in size.  So if you want to connect a different
+// kernel/IP make sure to place it between data master address
+// 0x0010_0000-0x001F_FFFF.
+
+#include <stdio.h>
+#include <stdlib.h>
+
+// This comes from the BSP and contains Nios V data cache flushing APIs
+#include <sys/alt_cache.h>
+
+// This comes from the BSP that was generated by software_build.h, it includes
+// the macros used to access the control/status registers of peripherals
+#include "io.h"
+
+// This comes from the BSP that was generated by software_build.h, it contains
+// information like address map, IRQ mapping, etc...
+#include "system.h"
+
+// including the kernel register map directly from the kernel build directory
+#include "../../../kernels/simple_dma/build/simple_dma.report.prj/include/register_map_offsets.hpp"
+
+// In bytes, must be a multiple of 4.  Keep it a small number to shorten the
+// simulation time and do not exceed the 1MB memory size (remember this code is
+// in there too). Once the DMA gets going this buffer will fly by fast so
+// setting it too high mostly affects the memory initialization and the
+// correctness check at the end once the DMA gets going this buffer will fly by
+// fast so setting it too high mostly affects the memory initialization and the
+// correctness check at the end.
+#define BUFFER_LENGTH 1024
```

Comment:
When using SYCL/C++ it's prefered to not use #define if their are more typesafe ways of solving the problem.   The usual solution is to use constexpr to define correct types of expressions and values.

Reply:
This is not SYCL code, this is C code for the softprocessor that will be used in combination with the SYCL code compiled in `simple_dma.cpp`.
`simple_dma.cpp` is the SYCL code that is given to our SYCL/C++ oneapi compiler. I don't think file is compiled by our compiler, and I think that this is restricted to C, which does not support `constexpr` I believe

Reply:
@yuguen-intel  is correct.